divi/src/masternode.cpp

```diff
@@ -868,6 +872,32 @@ bool CMasternodeBroadcast::CheckAndUpdate(int& nDos)
     return true;
 }
 
+const CBlockIndex* CMasternodeBroadcast::GetCollateralConfirmationBlock() const
+{
+    uint256 hashBlock;
+    CTransaction tx;
+    if (!GetTransaction(vin.prevout.hash, tx, hashBlock, true))
+        return nullptr;
+
+    const auto mi = mapBlockIndex.find(hashBlock);
+    if (mi == mapBlockIndex.end() || mi->second == nullptr)
+        return nullptr;
+
+    const CBlockIndex* conf;
+    {
+        LOCK(cs_main);
+        if (!chainActive.Contains(mi->second))
+            return nullptr;
+        conf = chainActive[mi->second->nHeight + MASTERNODE_MIN_CONFIRMATIONS - 1];
+    }
+
+    if (conf == nullptr)
+        return nullptr;
+
+    assert(conf->GetAncestor(mi->second->nHeight) == mi->second);
+    return conf;
```

Comment:
The height alone is not enough.  We need also the block time of the block (`conf->GetBlockTime()`) later in `CheckInputs`.  That is why the method returns the `CBlockIndex*` of when the collateral got 15 confirmations, rather than just the collateral height.